WordPress/openverse
# Allow using special characters in creator name and tags

## Fixes



Fixes #3791 by @AetherUnbound

## Description


This PR changes the way the path is parsed to extract the creator name to allow for URL-encoded special symbols like `/` and `?` within it. The creator name is also encoded when creating the app creator page link.
To enable testing this change locally, I added special symbols to the creator names sample data.

## Testing Instructions

Run the local API. 

Then, run the Nuxt app using the local API by running the app using `API_URL=http://localhost:50270 just frontend/run dev:only`

Enable `additional_search_views` on the `/preferences` page.
~~Go to `localhost:8443/audio/source/wikimedia_audio` and navigate to one of the results that have a slash in the creator name. From that single result page, click on the creator link to open the creator view - there should be no errors here.~~

Search for "Wish You Were Here" in audio. You should see an item with a question mark in the creator name. If you click on it, you should see that the name is encoded in the URL: http://localhost:8443/audio/source/jamendo/creator/The.madpix.project%3F/

If you search for "German term Vulkan" in audio, you should see an item with a `/` in the creator name. If you click on it, you should see the slash as is in the URL, however, this slash is actually encoded in the app code. You can get to this page using this URL: http://localhost:8443/audio/source/wikimedia_audio/creator/wiki%2FJeuwre/ 

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
